### PR TITLE
Globus Bulk Stager

### PR DESCRIPTION
**Pull request by dougbenjamin**
Pull request for Globus Bulk Stager

### CODE REVIEW

File: pandaharvester/harvesterstager/go_bulk_stager.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+                # escape since locked by another thread
+                msgStr = 'escape since locked by another thread'
+                tmpLog.debug(msgStr)
+                return None, msgStr
+            # refresh group information since that could have been updated by another thread before getting the lock
+            self.dbInterface.refresh_file_group_info(jobspec)
+            # get transfer groups again with refreshed info
+            groups = jobspec.get_groups_of_output_files()
+            # the dummy transfer ID is still there
+            if self.dummy_transfer_id in groups:
```
Comment (tmaeno) [inline pandaharvester/harvesterstager/go_bulk_stager.py]: I think that the lock is not released if this statement is False.

**Comment by dougbenjamin** [comment]
Hi

   The logic is correct. Initially self.have_db_lock = False when the object is instantiated.

So it should call - first time through because not False -> is True

self.have_db_lock = self.dbInterface.get_object_lock(self.dummy_transfer_id, lock_interval=120)

If the lock is set  then self.have_db_lock = True

otherwise self.have_db_lock = False if another process has the lock

If self.have_db_lock is false the logic is to exit.

This is no different than the code in dummy_bulk_stager.py from which the code is patterned.
https://github.com/PanDAWMS/panda-harvester/blob/master/pandaharvester/harvesterstager/dummy_bulk_stager.py#L31

         if dummy_transfer_id in groups:
             # lock for 120 sec
             locked = self.dbInterface.get_object_lock(dummy_transfer_id, lock_interval=120)
             if not locked:
                 # escape since locked by another thread
                 msgStr = 'escape since locked by another thread'
                 tmpLog.debug(msgStr)
                 return None, msgStr
             # refresh group information since that could have been updated by another thread before getting the lock
             self.dbInterface.refresh_file_group_info(jobspec)

**Comment by tmaeno** [comment]
OK, the dummy plugin was wrong. I've fixed it

https://github.com/PanDAWMS/panda-harvester/commit/d97490d79357500dcfd57b69bada287727c3b175

**Comment by dougbenjamin** [comment]
Please accept the merge request.
So I can easily re synchonize my fork and I will update the code and resubmit.

**Pull request merged**